gdsfactory/typings.py

```diff
@@ -45,6 +45,48 @@
 from gdsfactory.port import Port
 from gdsfactory.technology import LayerLevel, LayerStack
 
+
+STEP_DIRECTIVES = {
+    "x",
+    "y",
+    "dx",
+    "dy",
+}
+_STEP = Literal[
+    "x",
+    "y",
+    "dx",
+    "dy",
+]
+
+STEP_DIRECTIVES_ALL_ANGLE = {
+    "x",
+    "y",
+    "dx",
+    "dy",
+    "ds",
+    "exit_angle",
+    "cross_section",
+    "connector",
+    "separation",
+}
+
+_STEP_ALL_ANGLE = Literal[
+    "x",
+    "y",
+    "dx",
+    "dy",
+    "ds",
+    "exit_angle",
+    "cross_section",
+    "connector",
+    "separation",
+]
+
+
+Step = Dict[_STEP, float]
+StepAllAngle = Dict[_STEP_ALL_ANGLE, float]
```

Comment:
You're welcome to ignore this if it doesn't align with gdsfactory style / requirements / etc., but part of why I requested a dataclass for these is that it makes the fields more discoverable. If `Step` has an attribute `dy`, my IDE will autofill that attribute (and possibly display docs for it, if they exist). Keeping these as dicts, even with a `Literal` to enumerate key values, does not provide that benefit.

Reply:
see https://github.com/gdsfactory/gdsfactory/pull/1257/commits/0a717c4b08807086126dc0468c6a73d290a4fb58

Reply:
as long as it doesn't break the yaml flow, i'm ok